cloud-mgmt/internal/cloud-provider/persys/persys.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	capvv1beta1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+func main() {
+	// Set up the Kubernetes API client (see previous step)
+
+	// Register the CAPV types with the Kubernetes scheme
+	if err := capvv1beta1.AddToScheme(scheme.Scheme); err != nil {
+		fmt.Printf("Error adding CAPV to scheme: %v\n", err)
+		return
+	}
+	// Create a new CAPV VSphereCluster object
+	vsphereCluster := &capvv1beta1.VSphereCluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: capvv1beta1.GroupVersion.String(),
+			Kind:       "VSphereCluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-cluster",
+			Namespace: "default",
+		},
+		Spec: capvv1beta1.VSphereClusterSpec{},
+	}
+	// Create the CAPI kubernetes cluster object
+	cluster := &capi.Cluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: capi.GroupVersion.String(),
+			Kind:       "Cluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-cluster",
+			Namespace: "default",
+		},
+		Spec: capi.ClusterSpec{},
```

Comment:
we need to read the cluster spec from cluster-config.yaml and populate the data here.